Feature/event service

SongAhyeon01 opened the pull request:
## createReport 수정
- 새로운 리포트 저장 후, 기존 이벤트에 귀속 시킬 지 새로운 이벤트를 만들지 결정
- 거리 계산은 Calcul 클래스 이용
- if 문에서 거리 계산하여 가까운 이벤트를 찾으면 귀속 
& 해당 이벤트가 10개가 채워졌는지 확인하여 이벤트 활성화
- else 문에서 createEvent 호출

SongAhyeon01 committed:
Merge branch 'event/eventStatus' into feature/event-service

SongAhyeon01 committed:
feat : 이벤트 활성화 코드 생성

SongAhyeon01 committed:
feat : createReport 수정
- 거리 계산 후 1.0km 이내라면 해당 이벤트에 새로운 리포트 추가
- break 문으로 최초로 찾은 이벤트에 귀속됨
- 리포트 귀속 후 해당 이벤트에 속한 리포트가 10개라면 이벤트 활성화
- 1.0km 보다 멀다면 eventRequest(dto) 생성한 후 createEvent 호출

SongAhyeon01 committed:
Merge branch 'develop' into feature/event-service

bong-u reviewed (approved):
고생하셨습니다! createReport 함수는 로직이 복잡해서 추후에 테스트 코드를 짜봐야겠네요.

JJH123123123 reviewed (commented):
```

          EventCreateRequest eventRequest = new EventCreateRequest();
                eventRequest.setEvent_latitude(newReport.getLatitude());
                eventRequest.setEvent_longitude(newReport.getLongitude());

                eventService.createEvent(eventRequest);


부분에서 for 밖에 boolean checkExistEvent = false; 을 만들고 

for (Event event : events){
     if 거리 <= 1 {

    }
    // 거리가 >1 면 continue
}

if (!flag){
    // create Event 로 바꿔야 되지 않을까 싶습니다.
   // 이유는 해당 루프를 전부 탐색하고 나서, 
   // 해당 리포트와 1 km 범위 내에 이벤트가 존재하지 않을 때
  // 새로운 이벤트를 만들어주는 거잖아요? 
  // 그래서 for loop를 돌고 생성해야 되지 않을까 싶습니다.
}
```

SongAhyeon01 committed:
feat : createReport 수정2
- 새로운 이벤트 생성 요청 시 기본 상태 값을 Awaiting 으로 설정
- 이벤트에 귀속 or 이벤트 생성 루프 종료 후 이벤트에 10개가 채워졌는지를 확인 & 이벤트 활성화하는 루프 실행

SongAhyeon01 committed:
Merge remote-tracking branch 'origin/feature/event-service' into feature/event-service
